### ISSUE TITLE
reconsider the defaults of eslint config

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#14588]
things has changed since eslint was created in 2013. It is time to reconsider the defaults of eslint config.
I would suggest the changes:

1. sourceType: "script" => "module"

* native esm has been supported by latest modern browsers and node.js
* a large number of devs are writing esm, and using something like babel/ts to support older browser/node.js versions.

2. ecmaVersion: 5 => "latest"
we have often got issues like "unexpected token...." just because not config `ecmaVersion`. e.g. https://github.com/eslint/eslint/issues/11643
the change should be good for users & maintainers.

3. --ext: ".js" => ".js,.mjs,.cjs"

it is a breaking change, but it's very easy to restore the old behaviour
```js
{
  "parserOptions": {"ecmaVersion": 5, "sourceType": "script",}
}
```

if we agreed to the change, document & `eslint --init` should be updated to reflect the change, too.

**Comment by nzakas** [comment]
We can consider this when we move to flat config. Changing this for eslintrc is likely to break a lot of existing users’ configurations.

**Comment by btmills** [comment]
I like @nzakas' idea. Flat config gives us a chance to update defaults in a way that likely _reduces_ migration difficulty.

**Comment by aladdin-add** [comment]
agreed. I will update the OP later.

**Comment by nzakas** [comment]
I’ll handle this as I’m working on the new config.

**Comment by nzakas** [comment]
Fixed in #15185
